pilot/pkg/networking/plugin/plugin.go

```diff
@@ -54,6 +56,8 @@ func ModelProtocolToListenerProtocol(protocol config.Protocol) ListenerProtocol
 	case config.ProtocolTCP, config.ProtocolHTTPS, config.ProtocolTLS,
 		config.ProtocolMongo, config.ProtocolRedis, config.ProtocolMySQL:
 		return ListenerProtocolTCP
+	case config.ProtocolAuto:
```

Comment:
Let's start by adding the 'auto' to the 15006 split filter chain (with an install-time CLI option/env).